### DISCUSSION TITLE
Need help to understand these Cortex Compactor logs

### DISCUSSION THREAD

**Discussion by zulh-civo** [cortexproject/cortex#5156] [Q&A]
I'm not sure if this is related to [my previous questions](https://github.com/cortexproject/cortex/discussions?discussions_q=author%3Azulh-civo+). But I saw these logs a lot in the compactor container/pod which I don't really understand.

```
cortex-compactor-0 compactor level=warn ts=2023-02-17T02:18:58.820897578Z caller=updater.go:108 component=cleaner org_id=abc-cluster msg="skipped partial block when updating bucket index" block=01GR8JE1S7K6M441KYA8GNYM2V
...
cortex-compactor-0 compactor level=warn ts=2023-02-17T02:47:43.519367378Z caller=updater.go:200 component=cleaner org_id=def-cluster msg="skipped missing block deletion mark when updating bucket index" block=01GQ1573WZM74KPHK96ZMGE4JK
...
cortex-compactor-0 compactor level=error ts=2023-02-17T03:08:22.743055018Z caller=compactor.go:696 component=compactor msg="failed to compact user blocks" user=xyz-cluster err="mkdir data: file exists"
```

Can someone explain in simple forms what they actually mean and what should I do to fix them?

**Comment by friedrichg** [comment]
I have never seen that.

That error message comes from
https://github.com/cortexproject/cortex/blob/v1.14.1/pkg/compactor/compactor.go#L686-L698

I am sorry, you haven't provided enough logs to provide a definitive assertion.

There might be local corruption caused by your other issue. 

After you fix the other issue, try deleting all the files in the compactor directory and restart. The compactor does not need these files, it will download everything from object storage.
